### PR TITLE
Pet sitter profile in applications, Placeholders for no jobs, no applicants, no pet profiles and Refresh

### PR DESCRIPTION
**Pull request by yaminaik**

### CODE REVIEW

File: frontend/src/Dashboard.tsx
Comment (ManaliTanna) [inline frontend/src/Dashboard.tsx:1]: Looks like some changes have disappeared from the develop branch while resolving conflicts. Can you look into that. Fro example   

>                           <p>
>                               Location: {job?.location?.address ?? ""}, {job?.location?.city ?? ""},{" "}
>                               {job?.location?.zipcode ?? ""}
>                             </p>

File: frontend/src/Dashboard.tsx [hunk 1]
```diff
@@ -245,19 +247,11 @@ const Dashboard = () => {
...
-                          {myApplications?.location?.city ?? ""},{" "}
-                          {myApplications?.location?.zipcode ?? ""}
-                        </p>
-                        <p>Pay: ${myApplications.job.pay}</p>
-                        <p>Start: {myApplications.job.start}</p>
-                        <p>End: {myApplications.job.end}</p>
-                        <p className="font-bold mb-2">
-                          Application Status:{" "}
-                          {!myApplications.status ? "No Decision" : myApplications.status}
-                        </p>
```
Comment (ManaliTanna) [inline frontend/src/Dashboard.tsx:260]: Please revert to changes present in develop branch for "locations" and  "application status"

File: frontend/src/Dashboard.tsx [hunk 2]
```diff
@@ -112,13 +111,12 @@ const Dashboard = () => {
             const jobDetailsResponse = await axios.get(API_ROUTES.JOBS, {
               params: { id: myApplications.job },
             });
-            const jobDetail = jobDetailsResponse.data;
-
-            const locationDetailsResponse = await axios.get(
-              `${API_ROUTES.USER.LOCATION}?location_id=${jobDetail.location}`
-            );
```
Comment (ManaliTanna) [inline frontend/src/Dashboard.tsx:119]: We need to send location id in the request to get the correct response.

File: frontend/src/App.tsx [hunk 3]
```diff
@@ -105,11 +105,9 @@ const AppRouter = () => {
       <Route
         path="locations"
         element={
-          <ProtectedRoute>
-            <Home
-              authContext={{ onLogin, onRegister, passwordReset, authenticationState, ...rest }}
-            />
-          </ProtectedRoute>
+          <Home
```
Comment (kolharsam) [inline frontend/src/App.tsx:108]: Why did you remove `ProtectedRoute`?

File: frontend/src/ApplicationModal.tsx [hunk 4]
```diff
@@ -74,7 +77,7 @@ const ApplicationModal: React.FC<ApplicationModalProps> = ({ isOpen, onClose, ap
         );
         setTimeout(() => {
           window.location.reload();
-        }, 700);
+        }, 800);
```
Comment (kolharsam) [inline frontend/src/ApplicationModal.tsx:80]: Why are we doing this? Could we not make another API call to fetch the data again or something?

**Review by kolharsam — COMMENTED** [review]
please look into the comments made by @ManaliTanna as well. you're almost there!

File: frontend/src/App.tsx (hunk 3 above)
Comment (yaminaik) [reply frontend/src/App.tsx:108]: Because if it is there it will always redirect to home when refreshed.

File: frontend/src/ApplicationModal.tsx (hunk 4 above)
Comment (yaminaik) [reply frontend/src/ApplicationModal.tsx:80]: ya i tried that but i was not working i will check again

File: frontend/src/Dashboard.tsx (hunk 2 above)
Comment (yaminaik) [reply frontend/src/Dashboard.tsx:119]: ya i am reverting it to your changes. Somehow when i pulled and pushed again your changes  were overwritten by mine.

File: frontend/src/App.tsx (hunk 3 above)
Comment (kolharsam) [reply frontend/src/App.tsx:108]: But, are you sure only authenticated users are allowed to visit the homepage?

File: frontend/src/ApplicationModal.tsx (hunk 4 above)
Comment (kolharsam) [reply frontend/src/ApplicationModal.tsx:80]: If you haven't been able to figure it out, it's okay. Let's just move on with this.

**Comment by kolharsam** [comment]
Please make sure to merge `develop` before merging this PR.

**Review by kolharsam — APPROVED** [review]
